### PR TITLE
Enable TOTP for federated users

### PR DESCRIPTION
**Pull request by GANGANI**
## Purpose

With this PR changes, federated users will be able to enable TOTP when JIT provisioning is enabled.
Feature improvement issue: https://github.com/wso2/product-is/issues/11988

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java [hunk 1]
```diff
@@ -155,34 +164,54 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
...
             if (context.isRetrying()) {
                 retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
             }
-            boolean isTOTPEnabled = isTOTPEnabledForLocalUser(username);
-            if (log.isDebugEnabled()) {
-                log.debug("TOTP is enabled by user: " + isTOTPEnabled);
+            boolean isTOTPEnabled =false;
+            // Not required to check the TOTP enable state for the initial login of the federated users.
+            if (!isInitialFederationAttempt) {
+                isTOTPEnabled = isTOTPEnabledForLocalUser(UserCoreUtil.addDomainToName(username,
```
Comment (IsuraD) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java:208]: shall we change the variable name. it's confusing.

File: component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java [hunk 2]
```diff
@@ -311,21 +346,34 @@ protected void processAuthenticationResponse(HttpServletRequest request,
 
         diagnosticLog.info("Processing TOTP authentication response.");
         String token = request.getParameter(TOTPAuthenticatorConstants.TOKEN);
-        String username = context.getProperty("username").toString();
+        AuthenticatedUser authenticatingUser =
+                (AuthenticatedUser) context.getProperty(TOTPAuthenticatorConstants.AUTHENTICATED_USER);
+        String username = UserCoreUtil.addTenantDomainToEntry(authenticatingUser.getUserName(),
+                context.getTenantDomain());
+        username = UserCoreUtil.addDomainToName(username, authenticatingUser.getUserStoreDomain());
```
Comment (IsuraD) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java:353]: can't we use authenticatingUser.toFullQualifiedUsername

File: component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java (hunk 1 above)
Comment (GANGANI) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java]: Fixed https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/105/commits/9ea77f5434ce5d211d19c714fc2f09fadf192a88

File: component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java (hunk 2 above)
Comment (GANGANI) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java]: Fixed https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/105/commits/9ea77f5434ce5d211d19c714fc2f09fadf192a88

**Review by IsuraD — APPROVED** [review]

**Pull request merged**